wgpu-core/src/hub.rs

```diff
@@ -149,46 +149,59 @@ impl<T, I: id::TypedId> ops::IndexMut<id::Valid<I>> for Storage<T, I> {
 impl<T, I: id::TypedId> Storage<T, I> {
     pub(crate) fn contains(&self, id: I) -> bool {
         let (index, epoch, _) = id.unzip();
-        match self.map[index as usize] {
-            Element::Vacant => false,
-            Element::Occupied(_, storage_epoch) | Element::Error(storage_epoch, ..) => {
-                epoch == storage_epoch
-            }
+        match self.map.get(index as usize) {
```

Comment:
Could we avoid the double match here (and elsewhere and then have arms for Some(Occupied) etc